modules/top-level/test.nix

```diff
@@ -38,19 +38,44 @@ in
 
       nativeBuildInputs = [ config.finalPackage ];
 
-      # We need to set HOME because neovim will try to create some files
-      #
-      # Because neovim does not return an exitcode when quitting we need to check if there are
-      # errors on stderr
-      buildPhase = lib.optionalString cfg.runNvim ''
-        mkdir -p .cache/nvim
-
-        output=$(HOME=$(realpath .) nvim -mn --headless "+q" 2>&1 >/dev/null)
-        if [[ -n $output ]]; then
-          echo "ERROR: $output"
-          exit 1
-        fi
-      '';
+      # First check warnings/assertions, then run nvim
+      buildPhase =
+        let
+          inherit (config) warnings;
+          assertions = lib.nixvim.modules.getAssertionMessages config.assertions;
+
+          showErr =
+            name: lines:
+            lib.optionalString (lines != [ ]) ''
+              Unexpected ${name}:
+              ${lib.concatStringsSep "\n" (lib.map (v: "- ${v}") lines)}
+            '';
+
+          errors = lib.foldlAttrs (
+            err: name: lines:
+            err + showErr name lines
+          ) "" { inherit assertions warnings; };
+        in
```

Comment:
Would it be better to have all this in the `let` block at the head of the file? (Where we currently only have `cfg = config.test`)

Reply:
Up to you, I've started to find it easier to have the let blocks be close to their usage unless it's used through a module.